server/service/src/requisition/request_requisition/insert_program.rs

```diff
@@ -177,10 +197,144 @@ fn generate(
         .map(|line| line.item_id)
         .collect();
 
-    let requisition_line_rows =
+    let requisition_lines =
         generate_requisition_lines(ctx, &ctx.store_id, &requisition, program_item_ids)?;
 
-    Ok((requisition, requisition_line_rows))
+    let program_indicators = program_indicators(
+        connection,
+        Pagination::all(),
+        None,
+        Some(ProgramIndicatorFilter::new().program_id(EqualFilter::equal_to(&program.id))),
+    )?;
+
+    let indicator_values = generate_program_indicator_values(
+        connection,
+        &ctx.store_id,
+        &period_id,
+        program_indicators,
+        &other_party_id,
+    )?;
+
+    Ok(GenerateResult {
+        requisition,
+        requisition_lines,
+        indicator_values,
+    })
+}
+
+fn generate_program_indicator_values(
+    connection: &StorageConnection,
+    store_id: &str,
+    period_id: &str,
+    program_indicators: Vec<ProgramIndicator>,
+    customer_name_id: &str,
+) -> Result<Vec<IndicatorValueRow>, RepositoryError> {
+    let store_pref = get_store_preferences(connection, store_id)?;
+    let customer_store_ids: Vec<String> = NameRepository::new(connection)
+        .query(
+            store_id,
+            Pagination::all(),
+            Some(
+                NameFilter::new()
+                    .supplying_store_id(EqualFilter::equal_to(store_id))
+                    .is_customer(true),
+            ),
+            None,
+        )?
+        .into_iter()
+        .map(|name| name.name_row.id)
+        .collect();
+
+    let indicator_line_ids = program_indicators
+        .iter()
+        .flat_map(|program_indicator| {
+            program_indicator
+                .lines
+                .iter()
+                .map(|line| line.line.id.clone())
+        })
+        .collect::<Vec<String>>();
+    let indicator_column_ids = program_indicators
+        .iter()
+        .flat_map(|program_indicator| {
+            program_indicator
+                .lines
+                .iter()
+                .flat_map(|line| line.columns.iter().map(|column| column.id.clone()))
+        })
+        .collect::<Vec<String>>();
+
+    let values = IndicatorValueRepository::new(connection).query_by_filter(
+        IndicatorValueFilter::new()
+            .period_id(EqualFilter::equal_to(period_id))
+            .indicator_line_id(EqualFilter::equal_any(indicator_line_ids))
+            .indicator_column_id(EqualFilter::equal_any(indicator_column_ids))
+            .store_id(EqualFilter::equal_any(customer_store_ids.clone())),
+    )?;
+
+    let mut indicator_values = vec![];
+
+    for program_indicator in program_indicators {
+        for line in program_indicator.lines {
+            for column in line.columns.clone() {
+                let value = match column.value_type {
+                    Some(IndicatorValueType::String) => column.default_value.clone(),
```

Comment:
Not sure what we want to do with strings.. 🤷🏻  so just defaulting

Reply:
Sounds good